builder/vsphere/supervisor/step_watch_source.go

```diff
@@ -165,7 +171,7 @@ func (s *StepWatchSource) waitForVMReady(ctx context.Context, logger *PackerLogg
 				logger.Info("Source VM is NOT powered-on yet, continue watching...")
 			}
 
-		case <-time.After(time.Duration(s.Config.WatchSourceTimeoutSec) * time.Second):
+		case <-s.timeoutCh:
```

Comment:
So reading up a little on the use of time.After I found that this is actually not a preferred because the timer channel created by time.After is never closed if it never times out. So on a successful run here the function returns but the timer channel is not closed, which can lead to a memory leak. 

Given your usage the memory leak would not occur here but in the select statement within the getVMIngressIP function. If step.Run succeeds without issue that the timeoutCh is never closed. Which you can fix by using a time.NewTimer and classing timeoutCh.Stop() in the Run function. But I am proposing a slightly different solution using context.WithTimeout

```suggestion
		case <-ctx.Done():
```